c10/xpu/XPUFunctions.cpp

```diff
@@ -27,6 +28,23 @@ struct DevicePool {
   std::unique_ptr<sycl::context> context;
 } gDevicePool;
 
+void identifyDeviceType(
+    std::vector<sycl::device>& device_list,
+    std::vector<bool>& is_igpu,
+    std::vector<bool>& is_dgpu) {
+  for (const auto i : c10::irange(device_list.size())) {
+    const auto& device = device_list[i];
+    if (device.is_gpu()) {
+      // Generally, iGPUs and the host share a unified memory subsystem.
+      if (device.get_info<sycl::info::device::host_unified_memory>()) {
```

Comment:
It seems like a workaround rather than a formal solution. Meanwhile, it is implementation orientation.